core/modules/replica/tools/qserv-replica-calc-cs.cc

```diff
@@ -20,73 +20,26 @@
  * see <http://www.lsstcorp.org/LegalNotices/>.
  */
 
-/// qserv-replica-calc-cs.cc calculate and print a checksum of
-/// the specified file.
+/**
+ * qserv-replica-calc-cs.cc calculate and print a checksum of
+ * the specified file.
+ */
 
 // System headers
-#include <iostream> 
+#include <iostream>
 #include <stdexcept>
-#include <string> 
-#include <vector> 
 
 // Qserv headers
-#include "replica/FileUtils.h"
-#include "util/CmdLineParser.h"
-
-using namespace lsst::qserv;
-
-namespace {
+#include "replica/CheckSumApp.h"
 
-/// The name of an input file to be processed
-std::vector<std::string> fileNames;
+using namespace lsst::qserv::replica;
 
-/// USe the incremental engine if set
-bool incremental;
-
-/// The test
-void test() {
+int main(int argc, const char* const argv[]) {
     try {
-        if (incremental) {
-            replica::MultiFileCsComputeEngine eng(fileNames);
-            while (not eng.execute()) { ; }
-            for (auto const& name: fileNames) {
-                std::cout << name << ": " << eng.cs(name) << std::endl;
-            }
-        } else {
-            for (auto const& name: fileNames) {
-                std::cout << name << ": " << replica::FileUtils::compute_cs(name) << std::endl;
-            }
-        }
+        auto app = CheckSumApp::create(argc, argv);
+        return app->run();
     } catch (std::exception const& ex) {
-        std::cerr << ex.what() << std::endl;
-        std::exit(1);
+        std::cerr << "main()  the application failed, exception: " << ex.what() << std::endl;
     }
+    return 1;
```

Comment:
This is confusing (probably because of the diffs) as it looks like main always returns 1. Can you probably move this `return` inside `catch`? Same for other apps.

Reply:
Agreed